python/src/scipp/_utils/profile.py

```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+import os
+import time as _time
+
+
+def time(method):
+    """Decorator that prints function run time."""
+    def timed(*args, **kw):
+        indent = time.counter
+        time.counter += 1
+        # Not using print() since output may be swallowed in Jupyter notebooks
```

Comment:
Is this really the case? Where did you see this behaviour?

![os_write_jupyter](https://user-images.githubusercontent.com/8266281/91482281-1bfcf800-e8a6-11ea-9db8-6bd1846c2056.png)

Reply:
This is useful since sometimes `print` statement inside the code you are importing get swallowed in Jupyter notebooks.
The best example of this is adding a `print` statement inside a function connected to a matplotlib event: you see the output of the print neither in the notebook, nor in the terminal that is running the notebook.
See https://github.com/jupyter/notebook/issues/244

`os.write` allows you to print stuff in the terminal.